src/FileSystem-Core/DiskDirectoryEntry.class.st

```diff
@@ -0,0 +1,338 @@
+
+	^DateAndTime fromUnixTime: ((self statAttributes at: 12) ifNil: [ self statAttributes at: 11 ])
+]
+
+{ #category : #accessing }
+DiskDirectoryEntry >> creationUnixTime [
+	"Answer the receivers creation time.  If this is not available, answer the change time, i.e. the time at which the receivers metadata was last updated."
+
+	^(self statAttributes at: 12) ifNil: [ self statAttributes at: 11 ]
+]
+
+{ #category : #accessing }
+DiskDirectoryEntry >> deviceId [
+	"Answer the device ID of the receiver"
+	
+	^self statAttributes at: 4
+]
+
+{ #category : #'cache management' }
+DiskDirectoryEntry >> getAccessAttributes [
+	"Store the receivers access() attributes for future reference"
+
+	^accessAttributes := self primitives fileAttributes: self pathString mask: self accessAttributeMask
+]
+
+{ #category : #'cache management' }
+DiskDirectoryEntry >> getAllAttributes [
+	"Store all the receivers attributes for future reference"
+
+	| allAttributes |
+	
+	allAttributes := self primitives fileAttributes: self pathString mask: self allAttributeMask.
+	statAttributes := allAttributes at: 1.
+	accessAttributes := allAttributes at: 2.
+	(statAttributes at: 1) ifNotNil: 
+		[ statAttributes at: 1 put: (self primitives decode: (statAttributes at: 1)) ].
+
+]
+
+{ #category : #'cache management' }
+DiskDirectoryEntry >> getStatAttributes [
+	"Store the receivers stat() attributes for future reference"
+
+	statAttributes := self primitives fileAttributes: self pathString mask: self statAttributeMask.
+	(statAttributes at: 1) ifNotNil: 
+		[ statAttributes at: 1 put: (self primitives decode: (statAttributes at: 1)) ].
+	^statAttributes
+]
+
+{ #category : #accessing }
+DiskDirectoryEntry >> gid [
+	"Answer the UID of the owner of the receiver"
+	
+	^self statAttributes at: 7
+]
+
+{ #category : #inspecting }
+DiskDirectoryEntry >> gtInspectAttributesIn: composite [
+	<gtInspectorPresentationOrder: 2>
+
+	| attrDictionary |
+
+	attrDictionary := Dictionary new.
+	self class methodsDo: [ :method |
+		method methodNode arguments size = 0 ifTrue:
+			[ attrDictionary at: method selector put:
+				(method selector value: self) ] ].
+	^(attrDictionary gtInspectorItemsIn: composite)
+		title: 'Attributes';
+		yourself.
+]
+
+{ #category : #testing }
+DiskDirectoryEntry >> hasCreationTime [
+	"Answer a boolean indicating whether the receiver has a creation timestamp, or is just using the change timestamp."
+
+	^(self statAttributes at: 12) isNotNil
+]
+
+{ #category : #accessing }
+DiskDirectoryEntry >> inode [
+	"Answer the index number of the receiver"
+	
+	^self statAttributes at: 3
+]
+
+{ #category : #'testing - stat' }
+DiskDirectoryEntry >> isBlock [
+	"Answer true if the receiver is a block device"
+	
+	^self primitives modeIsBlock: self mode.
+]
+
+{ #category : #'testing - stat' }
+DiskDirectoryEntry >> isCharacter [
+	"Answer true if the receiver is a character device"
```

Comment:
What is a character device?

Reply:
Hi Cyril,

> What is a character device?

The example I'm aware of is /dev/urandom.  But not really having a better understanding, here's a couple of pages I found:

"A character device typically transfers data to and from a user application — they behave like pipes or serial ports, instantly reading or writing the byte data in a character-by-character stream. They provide the framework for many typical drivers, such as those that are required for interfacing to serial communications, video capture, and audio devices. "

- http://derekmolloy.ie/writing-a-linux-kernel-module-part-2-a-character-device/

https://unix.stackexchange.com/questions/37829/how-do-character-device-or-character-special-files-work also has quite a bit of information, e.g. that character devices don't support the normal seek and tell operations.